src/status_im/notifications/local.cljs

```diff
@@ -107,30 +107,44 @@
      :user-info notification
      :message   description}))
 
+(defn chat-by-message
+  [{:keys [chats]} {:keys [chatId from]}]
+  (if-let [chat (get chats chatId)]
+    (assoc chat :chat-id chatId)
+    (assoc (get chats from) :chat-id from)))
+
 (defn show-message-pn?
   [{{:keys [app-state multiaccount]} :db :as cofx}
-   {{:keys [message chat]} :body}]
-  (let [chat-id (get chat :id)
-        chat-type (get chat :chatType)]
-    (and
-     (or (= app-state "background")
-         (not (chat.models/foreground-chat? cofx chat-id)))
-     (or (contains? #{constants/one-to-one-chat-type
-                      constants/private-group-chat-type}
-                    chat-type)
-         (contains? (set (get message :mentions))
-                    (get multiaccount :public-key))))))
+   {{:keys [message]}           :body
+    {:keys [chat-id chat-type]} :chat}]
+  (and
+   (or (= app-state "background")
+       (not (chat.models/foreground-chat? cofx chat-id)))
+   (or (contains? #{constants/one-to-one-chat-type
+                    constants/private-group-chat-type}
+                  chat-type)
+       (contains? (set (get message :mentions))
+                  (get multiaccount :public-key)))))
 
 (defn create-message-notification
-  ([cofx notification]
-   (when (or (nil? cofx)
-             (show-message-pn? cofx notification))
-     (create-message-notification notification)))
-  ([{{:keys [message contact chat]} :body}]
-   (let [chat-type    (get chat :chatType)
-         chat-id      (get chat :id)
+  ([{:keys [db] :as cofx} {{:keys [message]} :body :as notification}]
+   (when-not (nil? cofx)
+     (let [chat         (chat-by-message db message)
+           contact-id   (get message :from)
+           contact      (get-in db [:contacts/contacts contact-id])
+           notification (assoc notification
+                               :chat chat
+                               :contact-id contact-id
+                               :contact contact)]
+       (when (show-message-pn? cofx notification)
+         (create-message-notification notification)))))
+  ([{{:keys [message]} :body
+     {:keys [chat-type] :as chat} :chat
+     {:keys [identicon]} :contact
+     contact-id :contact-id}]
+   (let [chat-id      (get message :chatId)
          contact-name @(re-frame/subscribe
```

Comment:
i know it was introduced not in this PR, but its a wrong usage of subscription, it can be used only in views , here all values should be derived from db value

Reply:
yep that's true, I was a bit lazy. I will fix this separately soon